### PR TITLE
Mfdeakin sandia/st archive/encapsulate datetime

### LINKED ISSUE #2162: use of python datetime in case_st_archive precludes dates beyond 9999
We don't need to use datetime here.

### PR DESCRIPTION
**Pull request by mfdeakin-sandia**
This removes the use of Python's datetime object from the short term archiver, enabling archiving of dates year >9999

Test suite: scripts_regression_tests
Test baseline: 
Test namelist changes: 
Test status: BFB

Fixes  #2162

User interface changes?: N

Update gh-pages html (Y/N)?: N

Code review: @jedwards4b @jgfouca

### CODE REVIEW

**Comment by mfdeakin-sandia** [comment]
I've run B_CheckCode and the LDSTA test, the rest of scripts_regression_tests are still in progress, but I expect no issues

File: scripts/lib/CIME/utils.py [hunk 1]
```diff
@@ -1589,3 +1589,213 @@ def run_bld_cmd_ensure_logging(cmd, arg_logger, from_dir=None):
 
 def get_batch_script_for_job(job):
     return job if "st_archive" in job else "." + job
+
+class date:
```
Comment (billsacks) [inline scripts/lib/CIME/utils.py]: I feel like this class warrants its own file.

Comment (mfdeakin-sandia) [reply scripts/lib/CIME/utils.py]: I don't disagree, though I'm a little wary of cluttering the scripts directory with a utility class. If you're okay with it for this, I've moved it to a file just called `date.py`.

Comment (billsacks) [reply scripts/lib/CIME/utils.py]: Thanks.

**Pull request merged**